File: src/main/java/kr/co/lunatalk/global/util/MemberUtil.java
package kr.co.lunatalk.global.util;

import kr.co.lunatalk.domain.member.domain.Member;
import kr.co.lunatalk.domain.member.repository.MemberRepository;
import kr.co.lunatalk.global.exception.CustomException;
import kr.co.lunatalk.global.exception.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
@RequiredArgsConstructor
public class MemberUtil {
	private final MemberRepository memberRepository;
	private final SecurityUtil securityUtil;

	@Transactional(readOnly = true)
	public Member getCurrentMember() {
		return memberRepository
			.findById(securityUtil.getCurrentMemberId())
			.orElseThrow(() -> new CustomException(ErrorCode.MEMBER_NOT_FOUND));
	}

	@Transactional(readOnly = true)
	public Member getMemberByMemberId(Long memberId) {
		return memberRepository
			.findById(memberId)
			.orElseThrow(() -> new CustomException(ErrorCode.MEMBER_NOT_FOUND));
	}

}

File: src/main/kotlin/kr/co/lunatalk/global/util/MemberUtil.kt
package kr.co.lunatalk.global.util

import kr.co.lunatalk.domain.member.domain.Member
import kr.co.lunatalk.domain.member.repository.MemberRepository
import kr.co.lunatalk.global.exception.CustomException
import kr.co.lunatalk.global.exception.ErrorCode
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class MemberUtil(
    private val memberRepository: MemberRepository,
    private val securityUtil: SecurityUtil
) {

    val currentMember: Member
        get() = findCurrentMember()

    @Transactional(readOnly = true)
    fun findCurrentMember(): Member =
        memberRepository
            .findById(securityUtil.getCurrentMemberId())
            .orElseThrow { CustomException(ErrorCode.MEMBER_NOT_FOUND) }

    @Transactional(readOnly = true)
    fun getMemberByMemberId(memberId: Long): Member =
        memberRepository
            .findById(memberId)
            .orElseThrow { CustomException(ErrorCode.MEMBER_NOT_FOUND) }
}
